Gopkg.toml

```diff
@@ -284,3 +284,7 @@ required = [ "github.com/apache/thrift/lib/go/thrift"]
 [[constraint]]
   name = "github.com/spf13/pflag"
   version = "1.0.1"
+
+[[constraint]]
+  name = "github.com/dexidp/dex"
```

Comment:
Instead of installing dex as yet another dependency, I think it would be better to retrieve the proto file (download, submodule, whatever) and generate the client ourselves.

This is the recommendation in the dex docs BTW: https://github.com/dexidp/dex/blob/master/Documentation/api.md#generating-clients

See an example for handling proto files in a "real life" application here: https://github.com/sagikazarmark/modern-go-application

Although this would be nice to have in this PR, postponing to a subsequent would also be acceptable by me.

Reply:
Good idea, however, I would push that into a separate PR.

Reply:
See #1810